vivisect/parsers/elf.py

```diff
@@ -504,6 +516,30 @@ def loadElfIntoWorkspace(vw, elf, filename=None, baseaddr=None):
                     # Data Items (eg. literal pool)
                     logger.info('mapping (NOTYPE) data symbol: 0x%x: %r', sva, dmglname)
                     data_ptrs.append(sva)
+        elif s.getInfoType() == Elf.STT_OBJECT:
+            symname = s.getName()
+            if addbase:
+                sva += baseaddr
+            if symname:
+                vw.makeName(sva, symname, filelocal=True, makeuniq=True)
+                valu = vw.readMemoryPtr(sva)
+                if not vw.isValidPointer(valu) and s.st_size == vw.psize:
+                    vw.makePointer(sva, follow=False)
+                else:
+                    if not valu:
```

Comment:
Yea, you're right in that most of this is replicated in makePointer with follow=True. I specifically didn't use that, since that kicks off a bunch of other analysis that isn't safe to run yet (it blows up in fun ways), but I still want these locations made first, so that other analysis modules know to not monkey with these and so I can set sizes and what not. But ELF isn't super specific about *what* these are supposed to be, so I do some guessing here.

It's not great and I hate the code for it. But I was drawing a blank on what this should have really been, and while ugly, this does cover a couple nice use cases like pointer tables/arrays of pointers being present.

Reply:
cool.  
can you drop a comment about that in the code (ie. the duplication and why).
also, does it make sense to rip out some of the code from analyzePointer and call it here and from analyzePointer?  no biggie if it doesn't make sense

Reply:
```suggestion
                    '''
                    Most of this is replicated in makePointer with follow=True. We specifically don't use that, since that kicks off a bunch of other analysis that isn't safe to run yet (it blows up in fun ways), but we still want these locations made first, so that other analysis modules know to not monkey with these and so I can set sizes and what not. 
                    while ugly, this does cover a couple nice use cases like pointer tables/arrays of pointers being present.
                    '''
                    if not valu:
```